src/Monad/Writer.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Widmogrod\Monad;
+
+use Widmogrod\Common;
+use Widmogrod\FantasyLand;
+use Widmogrod\Monoid\StringMonoid;
+
+class Writer implements FantasyLand\Monad
+{
+    const of = 'Widmogrod\Monad\Writer::of';
+
+    public static function of($value, $side = '')
+    {
+        return new static($value, $side);
+    }
+
+    /** @var mixed */
+    protected $value;
+
+    /** @var FantasyLand\Monoid $side */
+    private $side;
+
+    public function __construct($value, $side = '')
+    {
+        $this->value = $value;
+
+        if(is_string($side)) {
+            $this->side = new StringMonoid($side);
+        } else if($side instanceof FantasyLand\Monoid) {
+            $this->side = $side;
+        } else {
+            throw new \RuntimeException("String or Monoid needed.");
+        }
+    }
+
+    public function bind(callable $function)
+    {
+        $new = $function($this->value);
+        $new->side = $this->side->concat($new->side);
```

Comment:
Because `$new` is the same type as use and thus the visibility does not matter

>
Objects of the same type will have access to each others private and protected members even though they are not the same instances. This is because the implementation specific details are already known when inside those objects.
>

From http://php.net/manual/en/language.oop5.visibility.php